keel-core/src/main/kotlin/com/netflix/spinnaker/keel/caffeine/CacheFactory.kt

```diff
@@ -66,16 +88,50 @@ class CacheFactory(
         .recordStats()
     }
 
-  private fun <K, V, C: AsyncCache<K, V>> C.monitor(cacheName: String) =
+  private fun <K, V, C : AsyncCache<K, V>> C.monitor(cacheName: String) =
     CaffeineCacheMetrics.monitor(meterRegistry, this, cacheName)
 }
 
-fun <K, V> (suspend (K) -> V?).toAsyncCacheLoader() : AsyncCacheLoader<K, V> =
+fun <K : Any, V> (suspend (K) -> V?).toAsyncCacheLoader(): AsyncCacheLoader<K, V> =
   AsyncCacheLoader<K, V> { key, executor ->
     CoroutineScope(executor.asCoroutineDispatcher())
       .future { this@toAsyncCacheLoader.invoke(key) }
   }
 
+fun <K : Any, V> (suspend () -> Map<K, V>).toAsyncBulkCacheLoader(): AsyncCacheLoader<K, V> =
+  object : AsyncCacheLoader<K, V> {
+    override fun asyncLoad(key: K, executor: Executor): CompletableFuture<V> {
+      throw UnsupportedOperationException()
+    }
+
+    override fun asyncLoadAll(
+      keys: Iterable<K>,
+      executor: Executor
+    ): CompletableFuture<Map<K, V>> =
+      CoroutineScope(executor.asCoroutineDispatcher())
+        .future { this@toAsyncBulkCacheLoader.invoke() }
+  }
+
+/**
+ * An implementation of [AsyncLoadingCache] that _always_ uses [AsyncCacheLoader.asyncLoadAll] to
+ * populate the cache.
+ */
+private class AsyncBulkLoadingCache<K : Any, V>(delegate: AsyncLoadingCache<K, V>) :
+  AsyncLoadingCache<K, V> by delegate {
+  override fun get(key: K): CompletableFuture<V> = getAll(listOf(key)).thenApply { it[key] }
+
+  override fun get(key: K, mappingFunction: Function<in K, out V>): CompletableFuture<V> {
+    throw UnsupportedOperationException()
+  }
+
+  override fun get(
+    key: K,
+    mappingFunction: BiFunction<in K, Executor, CompletableFuture<V>>
+  ): CompletableFuture<V> {
+    throw UnsupportedOperationException()
+  }
```

Comment:
We never use these `get` variants so I didn't bother to implement them and I don't think it really makes sense to anyway.